### ISSUE TITLE
Fix merkle ledger interfaces

### ISSUE THREAD

**Issue by nholland94** [MinaProtocol/mina#1474]

**Comment by nholland94** [comment]
@imeckler This is still outstanding. We don't need to do it before mainnet, but I don't think this should be closed.

**Comment by nholland94** [comment]
Fixes a core abstractional issue in our codebase, which will make it much easier to address future optimizations around scaling max ledger size. It mainly makes things easier to work with, easier to debug, and enables future work (but not any work that is scheduled for Q2). It could be pushed off, but it needs to be done before we can address a number of ledger related issues.

Doing this fixes a core abstractional issue in our codebase. Not doing this makes it difficult to make changes to our ledger structures, which will be important in the future as we scale TPS and ledger sizes (TPS is relevant here because of the nature of how ledger masks work). It will also make the code easier to debug and read.

**Comment by deepthiskumar** [comment]
duplicate of #14671
